src/main/java/org/myrobotlab/swing/ProgramABGui.java

```diff
@@ -178,23 +168,74 @@ public void actionPerformed(ActionEvent event) {
 
   @Override
   public void subscribeGui() {
+    subscribe("publishRequest");
+    subscribe("publishText");
+    subscribe("publishOOBText");
   }
 
   @Override
   public void unsubscribeGui() {
+    unsubscribe("publishText");
+    unsubscribe("publishRequest");
+    unsubscribe("publishOOBText");
+  }
+
+  public void onText(String text) {
+    SwingUtilities.invokeLater(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          responseKit.insertHTML(responseDoc, responseDoc.getLength(), "<i><b>" + botName.getText() + "</b>: " + text.replaceAll("\\<.*?\\>", " ").trim() + "</i>", 0, 0, null);

```

Comment:
Looks like you're re-inventing html parsing here.  Please use HttpFilter.stripHtml static method.